src/Xrefcheck/System.hs

```diff
@@ -20,6 +23,14 @@ import qualified System.FilePath.Glob as Glob
 readingSystem :: IO a -> a
 readingSystem = unsafePerformIO
 
+-- | Heuristics to check whether we are running within CI.
+-- Check the respective env variable which is usually set in all CIs.
+withinCI :: Bool
+withinCI = case readingSystem $ lookupEnv "CI" of
```

Comment:
Do we really need to `unsafePerformIO` here? Why not `withinCI :: IO Bool`?

Reply:
I honestly don't see much difference between these two. `IO` has too many capital letters and was too tiresome to write. One more minor reason if that if one day I need to use it in another place, it may be easier to use if it is pure.

Reply:
I don't see a big difference between reading env variables and CLI parsing in this sense. You could as well read command line options or config file inside `unsafePerformIO`. Command line options don't change during program lifetime, but how often do you see them being read using `unsafePerformIO`? And I think `IO` is intentionally short to be easy and quick to type :roll_eyes:

Reply:
Meh, okay, I have something to oppose to this.